src/mpl2/src/hier_rtlmp.cpp

```diff
@@ -6064,6 +6066,36 @@ void HierRTLMP::correctAllMacrosOrientation()
   }
 }
 
+void HierRTLMP::setMacroPlacementFile(const char* file_name)
+{
+  macro_placement_file_ = file_name;
+}
+
+void HierRTLMP::writeMacroPlacement(const char* file_name)
+{
+  std::string filename = file_name;
```

Comment:
Why not

void HierRTLMP::writeMacroPlacement(const std::string& file_name)

Reply:
I thought I was having some sort of compilation problem using std::string, but, apparently, it was related to the C++ swig methods.

I'll change it.

I'll change in `setMacroPlacementFile(const char* file_name)` as well.

Reply:
You can use a string in swig as well but you need to have
```
%include <std_string.i>
```
in the .i